### PR TITLE
EH_DWARF: collect information for Action Table

### PR DESCRIPTION
**Pull request by WalterBright**
Need to collect information necessary for generating the Action Table in gcc_except_table in the back end.

### CODE REVIEW

**Comment by yebblies** [comment]
malloc/realloc are not GC friendly, it would be much better if you used `mem.*` instead.

**Comment by WalterBright** [comment]
I didn't use mem because it is free'd by the back end which doesn't use mem.

File: src/backend/cc.h [hunk 1]
```diff
@@ -427,6 +427,8 @@ typedef struct block
         struct
         {   Symbol *catchtype;          // one type for each catch block
             #define Bcatchtype BS.BIJCATCH.catchtype
```
Comment (andralex) [inline src/backend/cc.h]: Is this define really necessary? Placement is really odd, too.

Comment (WalterBright) [inline src/backend/cc.h]: Not really, but it's been there forever.

**Pull request merged**